src/TransactionBuilder.ts

```diff
@@ -95,24 +99,34 @@ export abstract class TransactionBuilder {
     }
 
     public build(): Transaction {
-        if (!this._inner.hasTransactionid()) {
-            this._inner.setTransactionid(newTxnId(this._client.operator.account));
-        }
+        let node: [string, AccountId] | undefined;
 
-        if (!this._inner.hasTransactionvalidduration()) {
-            this.setTransactionValidDuration(maxValidDuration);
-        }
+        if (this._client != null) {
+            if (!this._inner.hasTransactionid()) {
+                this._inner.setTransactionid(newTxnId(this._client.operator.account));
+            }
+
+            node = this._getNode()!;
+            if (!this._inner.hasNodeaccountid()) {
+                this.setNodeAccountId(node[ 1 ]);
+            }
 
-        const [ url, nodeAccountID ] = this._getNode();
-        if (!this._inner.hasNodeaccountid()) {
-            this.setNodeAccountId(nodeAccountID);
+            if (!this._inner.hasTransactionvalidduration()) {
+                this.setTransactionValidDuration(maxValidDuration);
+            }
         }
 
         this.validate();
 
-        const txn = new Transaction_();
-        txn.setBodybytes(this._inner.serializeBinary());
+        const txProto = new Transaction_();
+        txProto.setBodybytes(this._inner.serializeBinary());
 
-        return new Transaction(this._client, url, txn, this._inner, this._method);
+        return new Transaction(
+            txProto,
+            this._inner,
+            this._method,
+            this._client,
+            node == null ? node : node[ 0 ]
```

Comment:
This can be shortened to `node && node[0]`

Reply:
`node?.[0]` works in Typescript 3.7 (released)